orion/pkg/domains/session.go

```diff
@@ -0,0 +1,40 @@
+package domains
+
+import (
+	"database/sql"
+	"errors"
+	"regexp"
+	"time"
+)
+
+var TABLE_SESSIONS = "sessions"
+
+type Session struct {
+	Id        uint
+	CreatedAt time.Time    `db:"created_at"`
+	UpdatedAt time.Time    `db:"updated_at"`
+	DeletedAt sql.NullTime `db:"deleted_at"`
+	ClassId   string       `db:"class_id" json:"classId"`
+	StartsAt  time.Time    `db:"starts_at"`
+	EndsAt    time.Time    `db:"ends_at"`
+	Canceled  bool         `json:"canceled"`
+	Notes     string       `json:"notes"`
+}
+
+func (session *Session) Validate() error {
+	// Retrieves the inputted values
+	classId := session.ClassId
+	notes := session.Notes
+
+	// Class ID validation
+	if matches, _ := regexp.MatchString(REGEX_GENERIC_ID, classId); !matches {
```

Comment:
I think this is unneeded because if classId is not valid, the database will scream because it's not an existing classId. So I don't think we have to check classId here